pkg/cliplugins/workspace/plugin/kubeconfig.go

```diff
@@ -215,7 +215,20 @@ func (o *UseWorkspaceOptions) Run(ctx context.Context) error {
 		if err != nil {
 			return err
 		}
-		newServerHost = homeWorkspace.Spec.URL
+		config, err := o.ClientConfig.ClientConfig()
+		if err != nil {
+			return err
+		}
+		u, _, err := pluginhelpers.ParseClusterURL(config.Host)
+		if err != nil {
+			return fmt.Errorf("current URL %q does not point to a workspace", config.Host)
+		}
+		uh, err := url.Parse(homeWorkspace.Spec.URL)
+		if err != nil {
+			return fmt.Errorf("invalid home workspace URL %q: %w", homeWorkspace.Spec.URL, err)
+		}
+
+		newServerHost = u.Scheme + "://" + path.Join(u.Host, uh.Path)
```

Comment:
maybe worth a comment what we are doing here, namely: keep the old host, but append the path of the homeworkspace URL.